tests/integration/optlyplugins/test_composite_service.go

```diff
@@ -35,15 +35,9 @@ func (c *TestCompositeService) AddListeners(listeners map[string]int) {
 	if len(listeners) < 1 {
 		return
 	}
-	for listenerType, count := range listeners {
-		for i := 1; i <= count; i++ {
-			switch listenerType {
-			case "Decision":
-				c.OnDecision(c.decisionNotificationCallback)
-				break
-			default:
-				break
-			}
+	if count, ok := listeners[models.KeyDecision]; ok {
```

Comment:
is this change needed in `OnTrack` integration. if not please add proper description.